frontend/src/components/chart_panel.jsx

```diff
@@ -91,6 +92,66 @@ const ChartPanel = ({ settings, id, setPanels }) => {
             },
         });
 
+        settings.tooltip && newChart.subscribeCrosshairMove((param) => {
+            const tooltipRef = tooltip.current;
+            if (!param || !param.seriesData) {
+                tooltipRef.style.display = 'none';
+                return;
+            }
+
+            let seriesData = new Array();
+            param.seriesData.forEach((data, key, map) => {
+                if (!key.options().visible) return;
+                let color = key.options().color || key.options().topFillColor1 || key.options().lineColor || key.options().upColor || key.options().borderColor;
```

Comment:
```suggestion
                let color = [
                    key.options().color,
                    key.options().topFillColor1,
                    key.options().lineColor,
                    key.options().upColor,
                    key.options().borderColor
                ].some(x => x);
```
What do yout think of doing it like that?

Reply:
@eliza-eliza's approach is a more common and effective way to get the first available non-null value. It seems that the suggested approach should use .find(x => x);

Reply:
Thanks, yes, using `.find(x=>x)` is more appropriate